synapse/storage/databases/main/events_worker.py

```diff
@@ -729,12 +731,16 @@ async def get_missing_events_from_db() -> Dict[str, EventCacheEntry]:
 
         return event_entry_map
 
-    def _invalidate_get_event_cache(self, event_id: str) -> None:
-        self._get_event_cache.invalidate((event_id,))
+    async def _invalidate_get_event_cache(self, event_id: str) -> None:
+        await self._get_event_cache.invalidate((event_id,))
+        self._invalidate_local_get_event_cache(event_id)
```

Comment:
I assume that doing the invalidations in this order is important. It might be nice to add some comments explaining why and saying how this works.

Reply:
https://github.com/matrix-org/synapse/pull/13242/commits/b064024b392746f724dc0d66c1b96dafe9e2f285